crates/matrix-sdk/src/sliding_sync.rs

```diff
@@ -371,11 +491,70 @@ impl SlidingSyncBuilder {
     pub fn without_account_data_extension(mut self) -> Self {
         self.extensions
             .get_or_insert_with(Default::default)
-            .lock_mut()
             .get_or_insert_with(Default::default)
             .account_data = None;
         self
     }
+
+    /// Build the Sliding Sync
+    ///
+    /// if configured, load the cached data from cold storage
+    pub async fn build(self) -> Result<SlidingSync> {
+        self.build_no_cache().map_err(Error::SlidingSyncBuilder)?.build().await
+    }
+}
+
+/// The sliding sync instance
+#[derive(Clone, Debug)]
+pub struct SlidingSync {
+    /// Customize the homeserver for sliding sync only
+    homeserver: Option<Url>,
+
+    client: Client,
+
+    /// The storage key to keep this cache at and load it from
+    storage_key: Option<String>,
+
+    // ------ Internal state
+    pos: PosState,
+
+    /// The views of this sliding sync instance
+    pub views: ViewsList,
+
+    subscriptions: RoomsSubscriptions,
+    unsubscribe: RoomUnsubscribe,
+
+    /// The rooms details
+    rooms: RoomsMap,
+
+    /// keeping track of retries and failure cunts
+    failure_count: Arc<std::sync::atomic::AtomicU8>,
```

Comment:
For style consistency, can you replace the qualified path with a use of `AtomicU8`?